

<H1><center>Our Process</center></H1>
<h2>Overview</h2>
The ConSpeciFix process tests candidate genomes for species membership against a pre-processed species database. ConSpeciFix infers whether or not the candidate genome is engaging in gene flow with the selected species. It does so by returning to you an email detailing gene flow metrics (<i>h/m</i> ratio) estimated both with and without your candidate genome. Candidate genomes must be uploaded in the form of a multi-FASTA file including all (or some) genes of the genome (in nucleotides).
<h2>Our Database</h2>

<center>
	<a target="_blank" href="res/database.jpg">
<img src='res/database.jpg' width="100" height="159"></a>
<br>
<i>click to enlarge</i>
</center>
<br>

The database that provides the species samples for the ConSpeciFix analysis has been pre-processed for efficiency. The database includes all species available on NCBI with more than 15 genomes (last update: October 2017):
<br><br>
<ul>
<li>Download of all complete and annotated genomes from NCBI. Protein coding genes are extracted and converted from .fna and .gff files into the FASTA format
</li>
<li>Identification of orthologous gene families using <a target="_blank" href="https://www.drive5.com/usearch/">USEARCH</a> and <a target="_blank" href="https://www.micans.org/mcl/index.html">MCL</a>. 
</li>
<li>Definition of the core genome and alignment of core genes using <a target="_blank" href="http://mafft.cbrc.jp/alignment/software/">MAFFT</a>.
</li>
<li>Computation of the distance matrix between all genomes using <a target="_blank" href="https://sco.h-its.org/exelixis/web/software/raxml/index.html"> RAxML</a>.
</li>
<li>Application of our distance-based algorithm to estimate the ratio of homoplasic to non-homoplasic alleles (<i> h/m</i>) for randomly sampled combinations of genomes.</li>
<li>Representation of <i> h/m</i> ratios in the "control" graph for each species.
</li>
</ul>
This database has been built for all species on NCBI that meet the criteria listed above, and will be expanded and updated as more genomes and species become available. 

<h2>Candidate Genome Analysis</h2>

<center>
	<a target="_blank" href="res/upload.jpg">
<img src='res/upload.jpg' width="100" height="159"></a>
<br>
<i>click to enlarge</i>
</center>
<br>

Once a user has uploaded a candidate genome, it can be compared against the database for any species. This comparison process includes:
<br><br>
<ul>
<li>Identifying orthologous genes between the user's candidate genome and the core genome of the selected species in the database.
</li>
<li>The core genome of the species is updated to include the candidate genome.
</li>
<li>The distance matrix between genomes is re-calculated using RAxML.
</li>
<li>Estimation of <i> h/m</i> ratios for randomly sampled combinations of genomes, including the uploaded genome.
</li>
<li>Construction of the "test" graph comparing <i> h/m</i> ratios of the selected species with and without the candidate genome.
</li>
<li>An email is generated containing graphs and associated statistics that is sent to the user.</li>
</ul>

<br>
For details on implementation, please inspect our <a target="_blank" href="https://github.com/Bobay-Ochman/ConSpeciFix"> github.</a> For details about the method, please see:
<br>
<br>
<div class="well">
Bobay LM, Ochman H. Biological species are universal across life's domains. Genome Biol Evol. 2017. doi: 10.1093/gbe/evx026.
</div>

<style>
.column {
    float: left;
    width: 50%;
    padding: 10px;
    height: 477px; /* Should be removed. Only for demonstration */
}

/* Clear floats after the columns */
.row:after {
    content: "";
    display: table;
    clear: both;
}

/* Responsive layout - makes the two columns stack on top of each other instead of next to each other */
@media (max-width: 960px) {
    .column {
        width: 100%;
    }
}
</style>