

<H1><center>Interpreting Results</center></H1>
<br>
The email with your results will contain four files:
<ul>
	<li>An <i>h/m</i> ratio graph estimating gene flow between your candidate genome and randomly sampled strains of the selected species</li>
	<li>An <i>h/m</i> ratio graph estimating gene flow between randomly sampled strains of the selected species without your candidate genome</li>
	<li>A box plot comparing <i>h/m</i> ratios with and without your candidate genome</li>
	<li>The plain-text output of the ConSpeciFix program</li>
</ul>

Along with a message similar to:
<br>
<div class="well">
Hello!
<br>
<br>Here are the results of your comparison.
<br>
<br>Average Identity Score relative to other members of the species: ##.###%
<br>Number of core genes orthologous to your genome: ###
<br>Member of (Species_name) according to Exclusion Criterion*: <b>(yes/no)</b>
<br>
<br>*Please refer to Bobay & Ochman, GBE 2017
<br>
<br>Thanks,
<br>The ConSpeciFix Team
<br>
<br>This message is in regards to the file uploaded on (Full date and time)
<br>Species testing against: (Species_name)
</div>

Text and attachments of an email showing that the uploaded genome is a member of the selected species can be found <a target="_blank" href="res/email_member_of_species.zip"> here</a>, while components showing a negative indication can be found <a target="_blank" href="res/email_non_member_of_species.zip">here</a>. 

<h2> Beyond a yes / no answer</h2>
An <i>h/m</i> ratio graph for a collection of genomes all of which show evidence of gene flow looks similar to the one below. It has a positive slope, typically reaches a plateau, and the standard deviation is relatively small.
<br>
<center><img src="res/Bifidobacterium_bifidum.png" height = "400px" width = "400px"></center>
Every species graph additionally contains two horizontal red lines. These are the average (lower line) and highest observed (upper) <i> h/m</i> ratios for simulated genomes evolved <i>in silico</i> without recombination. If a species' <i>h/m</i> graph plateaus within these lines, it would be indicative of a clonal species (no significant signal of recombination has been detected).
<br>
The gallery of every <i>h/m</i> graph for each species processed and available on this server can be downloaded <a target="_blank" href="https://s3.us-east-2.amazonaws.com/conspecifix-data-bucket/figures.zip">here</a>.
<br>
<br>
However, if your candidate genome engages in gene flow at a much lower rate than the other members of the species, the graph will feature a sharp drop, as well as a wide standard deviation as seen below.
<br>
<center><img src="res/email_non/testGraph.jpg" height="400px" width = "400px"></center>
<br>
This drop is due to that, as the number of randomly sampled genomes grow larger in size, so does the likelihood they will contain the sexually isolated genome.
<br>
The box plot furthers this examination as, if the <i>h/m</i> values are similar between all samples not including the candidate genome, and all samples with the candidate genome, then they can be said to engage in gene flow at a similar rate and be considered members of the same species. 
<br>
<center><img src="res/email_member/boxPlot.jpg" height="400px" width = "400px"></center>
<br>
If, however, the box plots show very different values for the samples with established members of the species on their own versus the samples including the candidate genome, then we can conclude that these genomes do not recombine at similar rates and are thus not members of the same species.
<br>
<center><img src="res/email_non/boxPlot.jpg" height="400px" width = "400px"></center>
<br>
The results.txt file simply lists the results of the automated analysis, putting together in one group the strains that recombine with each other, and in a second those that don't recombine with those in the fist group. File names that begin with "GCF_<i>timestamp</i>" are database strains from NCBI. Your file has been renamed to begin "upload_", and will appear at the bottom of whichever list it is determined to be on.
<br>
<br>
