

<H1><center>Using ConSpeciFix</center></H1>
<h2>Purpose</h2>
This server provides access to two species-identification tools that we have developed. The first is the <b>ConSpeciFix</b> process, which takes as input a candidate genome, and tests for evidence of gene flow against a selected species from our database. The second is the <b>Exploratory Phase</b> which, when given a candidate genome, reports the most closely related species based on DNA identity based on 42 marker genes. This second process can be used to generate a predicted candidate species that can then be tested in the full ConSpeciFix Process. A full explanation of the ConSpeciFix Process can be found <a href="#!/process">here</a>.
<br><br>
The following instructions apply to both species identification tools hosted on this server, as they have been designed to be operated by the same interface. Select which tool you want to use when submitting your genome.
<br><br>

For multiple files, or to run this analysis locally, please see our <a target="_blank" href="http://www.github.com/Bobay-Ochman">GitHub</a>.
<h2> Preparing your data</h2>
The file you upload must adhere to the <a target="_blank" href="http://www.bioinformatics.nl/tools/crab_fasta.html"> FASTA </a> format. In order for us to analyze your genome, there are a few requirements imposed on the file beyond that of the default FASTA format. The file must:
<ul>
<li> Contain nucleotide data, not protein data.</li>
<li> Contain protein coding genes (unannotated genomes can not be submitted) </li>
<li> Contain a single genome for each submission.</li>
</ul>
Some examples of acceptable files (originating from NCBI) can be found here:
<ul>
<li> <a target="_blank" href="/res/exampleFiles/Acetobacter_pasteurianus.fa">Acetobacter pasteurianus</a> </li>
<li> <a target="_blank" href="/res/exampleFiles/Bacillus_megaterium.fa">Bacillus megaterium</a> </li>
<li> <a target="_blank" href="/res/exampleFiles/Vibrio_campbellii.fa">Vibrio campbellii</a> </li>
</ul>

<h2> Submitting an analysis</h2>
<ul>
	<li> <b> Upload your data: </b> begin by draging and droping your prepared FASTA file into the file upload field. Additionally, provide us with an email address where we can return the results of your analysis. This drag and drop will only work on Chrome and Firefox, so please use one of these browsers when utilizing this tool.</li> 
    <li> <b> Tell us what you know: </b> select the candidate species you would like to test your genome against. If you do not know which candidate species to test against, you can search for the most closely related species by selecting "Submit genome for Exploratory Phase". This process will identify the most likely candidate species through DNA similarity, after which you can submit your genome again for a full species classification based on gene flow. </li>
    <li> <b> Get Answers: </b> you will receive an email when the results are ready, usually in a matter of hours.</li>
    <li> <b> Interpret Results: </b> look <a href="/#!/results"> here</a> for some tips about interpreting the results emailed to you.
</ul>

<h2> Common Errors</h2>
 
Occasionally, your email from ConSpeciFix might contain a message reading "The file you have uploaded appears to not be in the proper format... An error log is listed below." Several categories of errors that can be reported:
<br>
<br>
<div class="well">
Error: too few genes to compare. Please separate into genes as per the FASTA format. Number of genes identified: ###
</div>

This occurs when ten or fewer genes are identified in your FASTA file. The ConSpeciFix process works best on complete genomes separated into genes. If you have more genomic data, append it to your FASTA file and submit the file again.

<br>
<br>
<br>
<div class="well">
Error: longest gene is > 30kbp. Are your genes annotated? Max len: ###
</div>

This occurs when a sequence of DNA identified as a gene in the FASTA file is larger than 30kbp. As ConSpeciFix is intended to function on prokaryote species, genes larger than this size are generally an indication that the FASTA file is not actually separated by genes, but rather contains unannotated segments of the genome. Information on how to accomplish this can be found online, and can normally be completed using tools such as GeneMark or Glimmer.
<br>
<br>
<br>

<div class="well">
Error: too many non-nucleotide character in file: #### non-nucleotide to #### total
</div>

This occurs when characters other than A, T, G, C, -, or N appear as more than 10% of your uploaded sequences (Gene annotation labels do not count toward this limit). This is typically due to the presence of protein sequences being used instead of nucleotide sequences.





